### ISSUE TITLE
Accuracy is worse in large model compared to medium model

### ISSUE THREAD

**Issue by nhan000** [Const-me/Whisper#61]
I understand that you mostly tuned this for Medium model, but at times large model is needed and I think you'd be interested in the result.

I attach the 25 second audio and the transcribed files in this zip file: [President.zip](https://github.com/Const-me/Whisper/files/11026864/President.zip)

When run the model locally on my PC, medium.en is better than large-v1 which in turns is better than large-v2 (now just named "large").

However, when I ran the file using the [demo on Hugging Face](https://huggingface.co/openai/whisper-large-v2), the result of large v2 is fantastic:

![image](https://user-images.githubusercontent.com/85216095/226571738-ae981ece-55e3-4a04-82ff-0efe190a3907.png)

So I think there's a bug for the large model? I'm running the latest version (1.10.1). Hope you could add support for the large model too. Thank you!

**Comment by nhan000** [comment]
thank you. I have no idea how to do that though

**Comment by Ivanmatthew** [comment]
You could convert a fine tuned model instead, it might help as it's a different format (h5) and has more training data. (same link as previous message)

**Comment by albino1** [comment]
For me, the large model is significantly worse than the medium and even tiny models.  I never knew what the reason was, so maybe a bad conversion explains it.

**Comment by vricosti** [comment]
I agree it seems worse also for me because when I try stream.cpp with the large models it can barely recognize some words

**Comment by nhan000** [comment]
I think a possible reason here is that my graphic card has 8 GB VRAM, and thus not enough to run the large model which requires 10 GB VRAM. For those who also don't have good result with the large model could you comment what VRAM you have.

**Comment by albino1** [comment]
I'm using an RTX 4080 with 16gb of VRAM.  I'm not sure of any way in Windows to isolate how much VRAM any specific application is using, but I'm definitely not running out as even the total VRAM usage is significantly below 16gb.  Also, I would say that the large model is worse even with short audio clips, which you wouldn't think would use much VRAM, but I've never really tested that.

**Comment by nhan000** [comment]
thanks for reporting that the issue still happens with 16GB VRAM. I have 3070Ti with 8 GB VRAM. 

I followed the instructions in [this video](https://www.youtube.com/watch?v=ABFqbY_rmEk) to install all individual components (Python, PyTorch, etc.) and ran Whisper via Command Prompt. I should be talking directly to the model now and have this memory error when running the large (but not the medium) model:

![image](https://user-images.githubusercontent.com/85216095/230822193-b23e1792-a070-41c5-9140-13bbb93ff9b3.png)

When I ran the model via the Google Colab Notebook the result was great (same as in the original post above).

**Comment by albino1** [comment]
Hmm, interesting, might just be a separate problem rather than the root cause of this issue, but who knows.

Maybe this bug would be better reported over on [whisper.cpp](https://github.com/ggerganov/whisper.cpp/) so that @ggerganov can either rebuild the broken large v1 model or build us a large v2 model so we can take advantage of that?

**Comment by vricosti** [comment]
https://github.com/ggerganov/whisper.cpp/issues/675

**Comment by albino1** [comment]
So until I read that post @vricosti linked to I thought `ggml-large.bin` and `ggml-large-v1.bin` were the same thing just renamed for simplicity as they're the exact same filesize.  I just re-ran the President tests and `ggml-large-v1.bin` is significantly better than `ggml-large.bin`:

Here's `ggml-large.bin`, which I now understand to actually be v2, and it still just says `[The President of the United States gives a speech.]`:

[President-ggml-large-bin.txt](https://github.com/Const-me/Whisper/files/11192815/President-ggml-large-bin.txt)

And here's `ggml-large-v1.bin`, which is obviously much better than v2, and has some better and some worse transcription than the medium model:

[President-ggml-large-v1-bin.txt](https://github.com/Const-me/Whisper/files/11192831/President-ggml-large-v1-bin.txt)

So, yeah, it seems like the large v2 model is fundamentally broken.

**Comment by nhan000** [comment]
@albino1 

> ggml-large.bin, which I now understand to actually be v2

yep that's correct. Now `large-v2` is just `large` now, and the old `large` model is now `large-v1`

So again, I just ran the same audio via google Colab and Hugging Face (large-v1 is [here](https://huggingface.co/openai/whisper-large) and large-v2 is [here](https://huggingface.co/openai/whisper-large-v2)). The code for Colab notebook is just 3 line 

```
!pip install git+https://github.com/openai/whisper.git
!sudo apt update && sudo apt install ffmpeg
!whisper "President.mp3" --model large #(or large-v1)
```
The results are clear and consistent: large-v2 is better than large-v1 and the transcripts are the same whether running the model via Google Colab or on HuggingFace.

`Large-v2`: As President of the United States, I rejoice at the first Pan-American Congress in Washington, D.C. I believe that with God's help, our two countries shall continue to live side by side in peace and prosperity.

`Large-v1`: As President of the United States, I would like to assert an American Congress in Washington, D.C. I believe that with God's help, our two countries shall continue to live side by side in peace and prosperity.

_It looks to me that **the online version is the correct ones** and there's something wrong with the offline model that we downloaded._ Even when comparing the result of the online `large-v1` above with the result of offline `large-v1` "[President-ggml-large-v1-bin.txt](https://github.com/Const-me/Whisper/files/11192831/President-ggml-large-v1-bin.txt)" posted by albino1, the transcript aren't the same (while the transcripts are the same whether using Google Collab or HuggingFace).

**Comment by sdevgill** [comment]
The large model (large-v2) works great with a 3090 using @Const-me implementation - it's actually even more accurate than the Open AI's whisper API! 

Also ran some tests using the local whisper with CUDA and PyTorch, and @Const-me implementation is about 40% faster - again, on the large model.
